onnx/test/optimizer_test.py

```diff
@@ -618,6 +618,37 @@ def test_fuse_arithmetic_into_batch_norm_multi_broadcast(self):  # type: () -> N
         assert list(optimized_model.graph.node[1].input) == ["X3", "A"]
         assert list(optimized_model.graph.node[2].input)[1] == "B"
 
+    def test_fuse_arithmetic_into_batch_norm_sub_first(self):  # type: () -> None
+        batch_norm = helper.make_node("BatchNormalization", ["X1", "X2", "X3", "X4", "X5"], ["Y"])
+        arith = helper.make_node("Sub", ["A", "Y"], ["Z"])
+        graph = self.helper_make_graph_fuse_arithmetic_into_batch_norm([batch_norm, arith])
+        optimized_model = self._optimized(graph, ["fuse_arithmetic_into_batch_norm"])
+
+        assert len(list(optimized_model.graph.node)) == 2
+        assert optimized_model.graph.node[0].op_type == "BatchNormalization"
+        assert optimized_model.graph.node[1].op_type == "Sub"
+        assert list(optimized_model.graph.node[1].input) == ["A", "Y"]
+
+    def test_fuse_arithmetic_into_batch_norm_dim_exceed(self):  # type: () -> None
+        batch_norm = helper.make_node("BatchNormalization", ["X1", "X2", "X3", "X4", "X5"], ["Y"])
+        arith = helper.make_node("Add", ["Y", "A"], ["Z"])
+        graph = helper.make_graph(
+            [batch_norm, arith],
+            "test",
+            [helper.make_tensor_value_info("X1", TensorProto.FLOAT, (1, 5, 3, 3)),
+             helper.make_tensor_value_info("X2", TensorProto.FLOAT, (5,)),
+             helper.make_tensor_value_info("X3", TensorProto.FLOAT, (5,)),
+             helper.make_tensor_value_info("X4", TensorProto.FLOAT, (5,)),
+             helper.make_tensor_value_info("X5", TensorProto.FLOAT, (5,)),
+             helper.make_tensor_value_info("A", TensorProto.FLOAT, (1, 1, 5, 1, 1))],
```

Comment:
what if the size of A were

- (5, 1, 1)
- (1, 5, 1, 1)
- (4, 5, 3, 3)
- (4, 5, 1, 1)

these are all valid graphs, which should disable this optimization, right? Because when they use an explicit Add operator, they use generalized numpy broadcasting, but when they are converted to use scale/bias, there is no broadcasting allowed

Reply:
@anderspapitto I got your point. Yes, it's right. Let me fix it.